### ISSUE TITLE
QoreProgram::parseCmdLineDefines() uses an API that performs a needless copy of a map

### ISSUE THREAD

**Issue by davidnich** [qorelanguage/qore#1099]
ex:

```
DLLEXPORT void parseCmdLineDefines(const std::map<std::string, std::string> defmap, ExceptionSink& xs, ExceptionSink& ws, int w);
```

in fact linking fails with g++ 5.4.0 and the newest binutils with this API:

```
qore-single-compilation-unit.o: In function `qore_main_intern(int, char**, int)':
/export/home/dnichols/src/el5/qore/git/qore/command-line.cpp:895: undefined reference to `QoreProgram::parseCmdLineDefines(std::map<std::string, std::string, std::less<std::string>, std::allocator<std::pair<std::string const, std::string> > >, ExceptionSink&, ExceptionSink&, int)'
collect2: error: ld returned 1 exit status
```

**Comment by pavelkveton** [comment]
@davidnich it looks like this was actually NOT fixed in 0.8.11.x, was it? can we change the milestone, then?

**Comment by davidnich** [comment]
@pavelkveton ok done
